code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm

```diff
@@ -92,10 +93,14 @@ GLOBAL_LIST_EMPTY(human_ai_brains)
 		lose_target()
 		return
 
-	var/possible_target = get_target()
-	if(possible_target)
-		lose_target()
```

Comment:
was this line not needed to make things work?

Reply:
Hmmm true. i forgot that we also leave signals on targets 😅

Reply:
Nevermind, we ONLY clear current target with lose_target() proc, which means "if ``current_target = null`` - we already have our signal unregistered", or else we have no reference to unregister from (and just set null to null)
https://github.com/Human-AI-copy/PvE-CM-HUMAN-AI/blob/78ff85845b939bb54c8dbc0d731cec51ab2ef682/code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm#L168-L173

Reply:
fair